src/UI/MainWindow.vala

```diff
@@ -76,8 +76,13 @@ public class Spreadsheet.UI.MainWindow : ApplicationWindow {
                         expression.sensitive = false;
                     }
                 });
-                sheet.focus_expression_entry.connect (() => {
-                    expression.grab_focus ();
+                sheet.focus_expression_entry.connect ((input) => {
+                    if (input != null) {
+                        expression.text += input;
+                    }
+                    expression.grab_focus_without_selecting ();
+                    // int.MAX so that it moves to the end of the buffer, whatever size it is
+                    expression.move_cursor (Gtk.MovementStep.BUFFER_ENDS, int.MAX, false);
```

Comment:
```suggestion
                    expression.move_cursor (Gtk.MovementStep.BUFFER_ENDS, expression.text.length, false);
```

I think `expression.text.length` instead of `int.MAX` would be more reasonable.